### PR TITLE
Turn segment

### PR DESCRIPTION
**Pull request by drdeaton**
Adds a segment to the Reckless controller for turns

### CODE REVIEW

**Comment by drdeaton** [comment]
wait why the fuck are there 40k lines changed

**Comment by drdeaton** [comment]
I fixed the CRLF endings to be LF endings. Please don't do that again

**Comment by Walk3rc04** [comment]
Ok, I didn’t do anything different on purpose

File: src/api/alg/reckless/turn_segment.cc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+  // Check Current angle/angular velocity compared to distance from target and
+  // set controller_state
+
+  // Remain in fullpower if we are currently in fullpower and we arent ready to
+  // advance
+
+  // edge case test if already at angle (having trouble in google test )
+  if (fabs(target_relative_original.convert(degree)) < 5.0) {
+    std::cout << "PASSED THE EDGE CASE :D" << std::endl;
```
Comment (dimi1729) [inline src/api/alg/reckless/turn_segment.cc:92]: remove print statements for merging

File: src/api/alg/reckless/turn_segment.cc [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  // Start slowdown if we are ready for that and we haven't already started
+  // harsh-braking
+  if (fabs(target_relative.convert(degree)) <=
+          fabs(current_state.vel.angular.convert(degree / second) *
+               coast_coeff) &&
+      controller_state != TurnState::BRAKE &&
+      controller_state != TurnState::COAST) {
+    std::cout << "setting COAST" << std::endl;
```
Comment (dimi1729) [inline src/api/alg/reckless/turn_segment.cc:103]: remove cout

File: src/api/alg/reckless/turn_segment.cc [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    controller_state = TurnState::COAST;
+  }
+  // Harsh-brake if we're at that point
+  if (fabs(target_relative.convert(degree)) <
+          fabs(current_state.vel.angular.convert(degree / second) *
+               harsh_coeff) &&
+      controller_state != TurnState::BRAKE) {
+    std::cout << "setting BRAKE" << std::endl;
```
Comment (dimi1729) [inline src/api/alg/reckless/turn_segment.cc:111]: remove cout

**Review by dimi1729 — CHANGES_REQUESTED** [review]

**Review by dimi1729 — APPROVED** [review]
